routing/road_access.hpp

```diff
@@ -12,21 +18,44 @@ namespace routing
 class RoadAccess final
 {
 public:
-  std::vector<uint32_t> const & GetPrivateRoads() const { return m_privateRoads; }
+  enum class Type : uint8_t
+  {
+    No,
+    Private,
+    Destination,
```

Comment:
Please add comments for every field. Especially for |Private| and |Destination| to show the difference.